src/components/AccordionList/AccordionListWrapper.tsx

```diff
@@ -0,0 +1,75 @@
+import { ChangeEvent, useCallback, useState } from "react";
+import AccordionList from "./AccordionList.tsx";
+import { IAccordion } from "../../models/IAccordion.ts";
+import { ACCORDION } from "../../data/accordion.ts";
+
+const AccordionListWrapper = () => {
+  const [searchFilter, setSearchFilter] = useState<string>("");
+  const [accordionData, setAccordionData] = useState<IAccordion[]>(ACCORDION);
+
+  const handleOpenItem = useCallback((id: number) => {
+    setAccordionData((prevState: IAccordion[]) => {
+      return prevState.map((item: IAccordion): IAccordion => {
+        return item.id === id
+          ? { ...item, isOpened: !item.isOpened }
+          : item.children.length > 0
```

Comment:
Мне кажется, такой вариант улучшит читабельность кода
```suggestion
          : item.children
```

Reply:
скорее наверное item.children.length, потому что суть в том, что массив children должен бытьн не пустой, а если item.children, то ты проверяешь на undefined, и даже если он пустой всегда будет true, потому что у каждого есть это поле

Reply:
Не сказал бы, я в своём коде использовал кусок кода, который привёл выше и всё работает, так что смею предположить, что оно всё таки будет выводить false